src/link/linkconfiguration.cpp

```diff
@@ -66,9 +66,38 @@ LinkConfiguration::Error LinkConfiguration::error() const
     return NoErrors;
 }
 
-//Todo: move to `operator QString()`
-QString LinkConfiguration::toString() const
+bool operator==(const LinkConfiguration& first, const LinkConfiguration& second)
+{
+    auto firstLinkconf = first.configurationStruct();
+    auto secondLinkconf = second.configurationStruct();
+    return (firstLinkconf.name == secondLinkconf.name)
+           && (firstLinkconf.type == secondLinkconf.type)
+           && (firstLinkconf.args == secondLinkconf.args)
+           ;
+}
+
+QDebug operator<<(QDebug d, const LinkConfiguration& other)
 {
     QString text(QStringLiteral("LinkConfiguration{Name: %1, LinkType: %2, Arguments: (%3)}"));
-    return text.arg(name(), QString::number(type()), args()->join(":"));
-}
\ No newline at end of file
+    d << text.arg(other.name(), QString::number(other.type()), other.args()->join(":"));;
+    return d;
+}
+
+QDataStream& operator<<(QDataStream &out, const LinkConfiguration linkConfiguration)
+{
+    out << linkConfiguration.configurationStructPtr()->name;
```

Comment:
if we didn't use ptr() would it make a copy?

Reply:
Yes, because it'll return a copy.